File: GrowthTracker/Entity/TaskSelection.cs
using System;

namespace GrowthTracker.API.Entity;

public class TaskSelection
{
    public int Id { get; set; }

    public Guid UserId { get; set; }
    public User User { get; set; } = null!;

    public int DailyTaskId { get; set; }
    public DailyTask DailyTask { get; set; } = null!;

    public DateTime SelectedAt { get; set; } = DateTime.UtcNow;
    public DateTime? CompletedAt { get; set; }
    public string Status { get; set; } = TaskSelectionStatus.Pending;
}

public static class TaskSelectionStatus
{
    public const string Pending = "Pending";
    public const string Active = "Active";
    public const string Completed = "Completed";
    public const string Skipped = "Skipped";
}

File: growth_tracker/lib/models/task_selection.dart
class TaskSelection {
  final int id;
  final String userId;
  final int dailyTaskId;
  final DateTime selectedAt;
  final DateTime? completedAt;
  final String status;

  TaskSelection({
    required this.id,
    required this.userId,
    required this.dailyTaskId,
    required this.selectedAt,
    this.completedAt,
    required this.status,
  });

  factory TaskSelection.fromJson(Map<String, dynamic> json) {
    return TaskSelection(
      id: json['id'] as int,
      userId: json['userId'] as String,
      dailyTaskId: json['dailyTaskId'] as int,
      selectedAt: DateTime.parse(json['selectedAt'] as String),
      completedAt: json['completedAt'] != null
          ? DateTime.parse(json['completedAt'] as String)
          : null,
      status: json['status'] as String,
    );
  }
}
